### PR TITLE
Wow, 4 way pipes.

### PR DESCRIPTION
**Pull request by Laboredih123**
[ATMOS][FEATURE]

## About the PR 
Adds a manifold that connects to, get this, ALL FOUR WAYS. Amazing stuff really. Doesn't add them to any map yet however, that's for the mappers to enjoy.

## Why's this needed? 
Be real nice to not have to mess with manifolds that have a mere 3 ways and therefore require more of them.

## Amaze thyself
image

## Changelog 

```changelog
(u)cringe
(*)Added a 4-way pipe, to be seen in Nanotrasen stations in the future!
```

### CODE REVIEW

File: code/modules/atmospherics/machinery/pipe.dm [hunk 1]
```diff
@@ -685,3 +685,114 @@
...
+		if(target.initialize_directions & get_dir(target,src))
+			src.node3 = target
+			break
+
+	for(var/obj/machinery/atmospherics/target in get_step(src, EAST))
+		if(target.initialize_directions & get_dir(target,src))
+			src.node4 = target
+			break
+
+	var/turf/T = src.loc			// hide if turf is not intact
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:797]: lot of tabs on the comment?

File: code/modules/atmospherics/machinery/pipe.dm [hunk 2]
```diff
@@ -685,3 +685,114 @@
...
+				parent.dispose()
+			parent = null
+		node3 = null
+
+	if(reference == node4)
+		if(istype(node4, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
+			parent = null
+		node4 = null
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:765]: 4 reference checks, but reference can only be one of the nodes? (Meaning could maybe do else ifs)

Also 4 parent checks, could maybe be made into 1?

File: code/modules/atmospherics/machinery/pipe.dm [hunk 3]
```diff
@@ -685,3 +685,114 @@
...
+	node3?.disconnect(src)
+	node4?.disconnect(src)
+	parent = null
+	..()
+
+/obj/machinery/atmospherics/pipe/quadway/disconnect(obj/machinery/atmospherics/reference)
+	if(reference == node1)
+		if(istype(node1, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:742]:
```suggestion
			parent?.dispose()
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 4]
```diff
@@ -685,3 +685,114 @@
...
+			parent = null
+		node1 = null
+
+	if(reference == node2)
+		if(istype(node2, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:749]:
```suggestion
			parent?.dispose()
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 5]
```diff
@@ -685,3 +685,114 @@
...
+			parent = null
+		node2 = null
+
+	if(reference == node3)
+		if(istype(node3, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:756]:
```suggestion
			parent?.dispose()
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 6]
```diff
@@ -685,3 +685,114 @@
...
+			parent = null
+		node3 = null
+
+	if(reference == node4)
+		if(istype(node4, /obj/machinery/atmospherics/pipe))
+			if (parent)
+				parent.dispose()
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:763]:
```suggestion
			parent?.dispose()
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 7]
```diff
@@ -685,3 +685,114 @@
...
+
+	if(!(src.node1 && src.node2 && src.node3 && src.node4))
+		parent.mingle_with_turf(loc, 70)
+
+/obj/machinery/atmospherics/pipe/quadway/disposing()
+	node1?.disconnect(src)
+	node2?.disconnect(src)
+	node3?.disconnect(src)
+	node4?.disconnect(src)
+	parent = null
```
Comment (FlameArrow57) [inline code/modules/atmospherics/machinery/pipe.dm:735]:
```suggestion
	src.node1?.disconnect(src)
	src.node2?.disconnect(src)
	src.node3?.disconnect(src)
	src.node4?.disconnect(src)
	src.parent = null
```
Generally in file too, may suggest using src.

**Comment by Laboredih123** [comment]
yay a review

File: code/modules/atmospherics/machinery/pipe.dm (hunk 2 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery/pipe.dm]: added else if

**Review by FlameArrow57 — COMMENTED** [review]
Idk how atmos/pipe code backend works, but no more comments from me, seems good

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**